### PR TITLE
Update S3 source with JSON

### LINKED ISSUE #738: Change required: add json support for s3 data ingestion
### Related code PR

https://github.com/risingwavelabs/risingwave/blob/main/e2e_test/s3/run.py#L22-L35

The kernel has tested S3 source with JSON format for a while.

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.risingwave.dev/docs/current/data-ingestion/

Add `JSON` in Amazon S3's format

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info
- **Description**: 
Update S3 source to reflect JSON support.

- **Preview**: 
[ Paste the preview link to the edited page here. ]

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/blob/main/e2e_test/s3/run.py#L22-L35

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/738

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/create-source/create-source-s3.md [hunk 1]
```diff
@@ -111,5 +120,28 @@ CREATE TABLE s(
...
+    age int,
+    mark int,
+) WITH (
+    connector = 's3',
+    match_pattern = '%Ring%*.ndjson',
+    s3.region_name = 'ap-southeast-2',
+    s3.bucket_name = 'example-s3-source',
+    s3.credentials.access = 'xxxxx',
+    s3.credentials.secret = 'xxxxx',
+     s3.endpoint_url = 'https://s3.us-east-1.amazonaws.com'
```
Comment (lmatz) [inline docs/create-source/create-source-s3.md]: one space unaligned 😄

**Review by lmatz — APPROVED** [review]

**Review by bernscode — APPROVED** [review]

File: docs/create-source/create-source-s3.md [hunk 2]
```diff
@@ -15,10 +15,13 @@ WITH (
...
    connector_parameter='value', ...
 )
-ROW FORMAT csv [WITHOUT HEADER] DELIMITED BY ','; 
+ROW FORMAT data_format
+[WITHOUT HEADER] [DELIMITED BY 'delimiter']; 
 ```
 
-
+:::info
+For CSV data, you must specify what the delimiter is with the `DELIMITED BY` clause.
```
Comment (hengm3467) [inline docs/create-source/create-source-s3.md]: For conciseness, I'd suggest rewording to: ... specify the delimiter in the `DELIMITED BY` clause...

**Review by hengm3467 — APPROVED** [review]
Otherwise looks good to me. Thanks.
